test/hotspot/jtreg/compiler/c2/aarch64/TestFloat16Replicate.java

```diff
@@ -0,0 +1,124 @@
+/* Copyright (c) 2025, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+* @test
+* @bug 8361582
+* @summary Ensure the correct backend replicate node is being generated for
+*          half precision float constants on >16B SVE machines
+* @modules jdk.incubator.vector
+* @library /test/lib /
+* @run main/othervm compiler.c2.aarch64.TestFloat16Replicate
+*/
+
+package compiler.c2.aarch64;
+
+import compiler.lib.ir_framework.*;
+import java.util.Arrays;
+import jdk.incubator.vector.Float16;
+import jdk.test.lib.*;
+
+import static java.lang.Float.*;
+import static jdk.incubator.vector.Float16.*;
+
+public class TestFloat16Replicate {
+    private static short[] input;
+    private static short[] output;
```

Comment:
This might give things even more chance to vectorize? Not sure, feel free to ignore.

```suggestion
    private static final short[] INPUTE;
    private static final short[] OUTPUT;
```

Reply:
I hope it's ok to not add these changes to the code. The loops are getting vectorized fine and the tests do pass on aarch64 and x86. I will consider this if there's any issue with auto vectorization in the futurre. Thanks

Reply:
I don't think that making the arrays final will make any difference for vectorization.